### DISCUSSION TITLE
bulk insert, ignoring existing keys (on conflict do nothing)

### DISCUSSION THREAD

**Discussion by jorgt** [supabase/supabase#6804] [Questions] [answered]
Hello, 

Given a table with a UUID V5 key, is there an option in the library to bulk insert entries, ignoring any existing entry with a key? I believe the SQL for it would be to `ON CONFLICT (ID) DO NOTHING`. 

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
Does upsert with ignoreDuplicates not do what you describe?
https://supabase.com/docs/reference/javascript/upsert

**Reply by MAXOUXAX** [reply]
I have this code:
```ts
const { error: errorNew } = await supabase.from('table').upsert(some_object, {
  onConflict: 'column_which_causes_the_error',
  ignoreDuplicates: true
});
```

I get this error:
```
{
  code: '42P10',
  details: null,
  hint: null,
  message: 'there is no unique or exclusion constraint matching the ON CONFLICT specification'
}
```

**Reply by steve-chavez** [reply]
@MAXOUXAX You need a unique key or primary key for ON CONFLICT to work. This is documented on PostgreSQL docs: https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT

**Reply by steve-chavez** [reply]
> Does upsert with ignoreDuplicates not do what you describe?

Yes, and behind the scenes it actually does a `ON CONFLICT (ID) DO NOTHING`.

**Reply by MAXOUXAX** [reply]
> @MAXOUXAX You need a unique key or primary key for ON CONFLICT to work. This is documented on PostgreSQL docs: https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT

The column name I used is not a primary key but the unique option is checked, so I'm wondering why it didn't work.

**Reply by steve-chavez** [reply]
Maybe try it with plain SQL on the SQL Editor, that could clear things up.

**Comment by MAXOUXAX** [comment]
This is much needed ⚡
